mirrord/layer/src/connection.rs

```diff
@@ -48,17 +49,22 @@ pub(crate) async fn connect(
     let progress = TaskProgress::new("agent initializing...");
 
     let agent_api = if let Some(address) = &config.connect_tcp {
-        let connection = Connection(address);
-
-        let stream = connection
-            .create_agent(&progress)
+        Connection(address)
+            .connect(&progress)
             .await
-            .unwrap_or_else(|_| panic!("Failed to connect to TCP socket {address:?}"));
-
-        connection
-            .create_connection(stream)
+            .unwrap_or_else(|err| handle_error(err, config))
+    } else if config.operator.enabeld && let Some((operator_api, operator_ref)) =
+        OperatorApiDiscover::discover_operator(config, &progress).await
+    {
+        operator_api
+            .create_connection(operator_ref)
             .await
             .unwrap_or_else(|err| handle_error(err, config))
+    } else if let Some(addr) = &config.operator.addr {
```

Comment:
when will you be able to connect to operator like this? is this for testing purposes or user use case?